packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -75,6 +75,13 @@ export interface IBucket extends IResource {
    */
   addToResourcePolicy(permission: iam.PolicyStatement): void;
 
+  /**
+   * Adds the provided bucket as a destination bucket for this bucket server access logs.
+   * @param destinationBucket Bucket in which the server access logs will be saved in.
+   * @param optional logFilePrefix of the access logs.
+   */
+  enableServerAccessLogs(destinationBucket: IBucket, logFilePrefix?: string): void;
```

Comment:
This can only be done on owned buckets and therefore does not belong to `IBucket` (which applies both owned and imported buckets).

Also, rename to `addServerAccessLogs` and add an option to `BucketProps` that allows configuring this when defining the bucket. Probably requires something like `s3.AccessLogOptions` struct that combines the bucket and the prefix.